File: projects/telegram/java/service/user/UserService.java
package jk.uz.pdp.service.user;

import jk.uz.pdp.model.User;
import jk.uz.pdp.service.BaseService;

import java.util.ArrayList;

public interface UserService extends BaseService<User> {
    ArrayList<User> users = new ArrayList<>();

    int setSMSCode(int smsCode, String phoneNumber);

    User signIn(String phoneNumber, int smsCode);

    ArrayList<User> searchByName(String name);
}

File: projects/telegram/service/user/user-service.ts
import { User } from "../../entities/user";
import { BaseService } from "../base-service";

export interface UserService extends BaseService<User> {
	users: User[];
	setSMSCode(smsCode: number, phoneNumber: string): number;
	signIn(phoneNumber: string, smsCode: number): User;
	searchByName(name: string): User[];
	getUsersByIds(ids: number[]): User[];
}
